internal/controllers/objectsetphases/phase_reconciler_test.go

```diff
@@ -0,0 +1,733 @@
+package objectsetphases
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"package-operator.run/apis/core/v1alpha1"
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+	"package-operator.run/package-operator/internal/controllers"
+	"package-operator.run/package-operator/internal/testutil"
+)
+
+func TestPhaseReconciler_TeardownPhase(t *testing.T) {
+	t.Run("already gone", func(t *testing.T) {
+		dynamicCache := &dynamicCacheMock{}
+		ownerStrategy := &ownerStrategyMock{}
+		r := &PhaseReconciler{
+			dynamicCache:  dynamicCache,
+			ownerStrategy: ownerStrategy,
+		}
+		owner := &unstructured.Unstructured{}
+
+		ownerStrategy.
+			On("SetControllerReference", mock.Anything, mock.Anything, mock.Anything).
+			Return(nil)
+
+		dynamicCache.
+			On("Get", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+			Return(errors.NewNotFound(schema.GroupResource{}, ""))
+
+		ctx := context.Background()
+		done, err := r.TeardownPhase(ctx, owner, corev1alpha1.ObjectSetTemplatePhase{
+			Objects: []corev1alpha1.ObjectSetObject{
+				{
+					Object: runtime.RawExtension{},
+				},
+			},
+		})
+		require.NoError(t, err)
+		assert.True(t, done)
+	})
+
+	t.Run("delete confirms", func(t *testing.T) {
```

Comment:
Could we think of a different name instead of "delete confirms". This is testing the case where the call to delete shows us that the object is already gone, right?

Reply:
yes exactly, changing to: `already gone when deleting`?

Reply:
Yeah, maybe `already gone on delete`?